lib/extract-templates.js

```diff
@@ -0,0 +1,135 @@
+import { parseTemplates } from 'ember-template-imports/lib/parse-templates.js';
```

Comment:
I think we actually need to do some utility extraction over in ember-template-imports. `parseTemplates` is trying to find all possible matches, but it doesn't care at all about the semantic meaning of them. 

We probably need to be using some extraction of `process-embedded-templates`, which leverages `parse-static-imports` so that we can make sure that we only process *specific* imported tagged template expressions.

Reply:
Sounds like a good thing to do in a followup, but I don't thing it should block this PR.

(getting a solid test suite is most important, so that such a cleanup can be done safely!)

Reply:
No, if we land this with `parseTemplates` we would be attempting to lint _all_ tagged template expression contents as if they were handlebars templates. This is fundamentally wrong, and a very large blocker.

Reply:
Code like:

```js

export default class Foo extends Component {
  indentedThing = stripIndent`
    Hahaha, this is just a plain string. Definitely not a template.
  `
}
```

That `stripIndent` will be linted by ember-template-lint as if it were a template.

Reply:
ah, I see. so parseTemplates is _just way wrong_ atm.

Reply:
Ya, it's very wrong for what _we_ want to do, but inside `ember-template-imports` it is used by `process-embedded-templates` properly as far as I can tell.